varinfo/umm_var.py

```diff
@@ -338,3 +341,53 @@ def get_json_serializable_value(input_value: Any) -> Any:
         output_value = input_value
 
     return output_value
+
+
+def publish_umm_var(collection_id: str,
+                    umm_var_dict: Dict,
+                    token: str,
+                    cmr_env: CMR_UAT) -> str:
+    """" Publish a single UMM-Var entry to CMR given:
+        * collection_id: a collection's concept_id
+        * umm_var_dict: a dictionary of a single UMM-Var entry for a collection
+        * token: Earthdata Login (EDL) bearer_token
+        * cmr_env/mode: CMR environments (OPS, UAT, and SIT)
+    For a successful publication response all of these fields must be entered
+
+    """
+    # Required UMM-Var headers for ingestiong variable entries
+    headers_umm_var = {
+        'Content-type': 'application/vnd.nasa.cmr.umm+json;version='
+        + f'{umm_var_dict["MetadataSpecification"]["Version"]}',
+        'Authorization': f'Bearer {token}',
+        'Accept': 'application/json'}
+
+    url_endpoint = (cmr_env.replace('search', 'ingest') + 'collections/'
+                    f'{collection_id}/variables/{umm_var_dict["Name"]}')
+
+    response = requests.put(url_endpoint,
+                            json=umm_var_dict,
+                            headers=headers_umm_var,
+                            timeout=10)
+    # Check the status_code of the response
+    # Return the variable concept-id if response is 200
+    if response.status_code == 200:
+        return response.json()['concept-id']
+    # Return the response if there was an error (response != 200)
+    return response
+
+
+def publish_all_umm_var(collection_id: str,
+                        all_umm_var_dict: Dict,
+                        token: str,
+                        cmr_env: CMR_UAT) -> List:
+    """ Publish all UMM-Var entries associated with a collection to CMR given:
+        * collection_id: a collection's concept_id
+        * all_umm_var_dict: a nested dictionary containing
+            dictionaries of all UMM-Var entries for a collection
+        * token: Earthdata Login (EDL) bearer_token
+        * cmr_env/mode: CMR environments (OPS, UAT, and SIT)
+    For a successful publication response all of these fields must be entered
+    """
+    return [publish_umm_var(collection_id, umm_var, token, cmr_env)
+            for umm_var in all_umm_var_dict.values()]
```

Comment:
I think I helped out with this list comprehension, so this is probably a criticism of my own code-golf...

The one thing that seems missing from this output is a link from the CMR response (concept ID or error message) to the actual UMM-Var JSON itself. Maybe that just means that this should be a dictionary comprehension instead...

```
return {variable_name: publish_umm_var(collection_id, variable_json, token, cmr_end)
        for variable_name, variable_json in all_umm_var_dict.items()}
```

If that sounds good, the return type hint would update to a `Dict`, too.